lib/translate.tsx

```diff
@@ -66,44 +72,83 @@ export function t(key: string, lang: string) {
     return '{missing_translation_key}'
   }
 
-  let result = get(translations, `${lang}.${key}`)
+  let translation = get(translations, `${lang}.${key}`)
 
-  if (!result) {
+  if (!translation) {
     return `{${key}}`
   }
 
-  if (result.indexOf('</Tooltip>') === -1) {
-    return result
+  if (
+    translation.indexOf('</Tooltip>') === -1 &&
+    translation.indexOf('</Link>') === -1
+  ) {
+    return translation
   }
 
-  const parts = []
-  let match
-  let lastIndex = 0
-
-  while ((match = tooltipRegex.exec(result))) {
-    const tooltipHtml = match[0]
-    const tkey = tooltipHtml.match(contentRegex)[1]
+  let result = []
 
-    const hrefMatch = tooltipHtml.match(hrefRegex)
-    const href = hrefMatch?.length > 0 ? hrefMatch[1] : null
+  result = injectComponent([translation], ComponentType.Link, lang)
+  result = injectComponent(result, ComponentType.Tooltip, lang)
```

Comment:
I have a code architecture-related question. 

Since you are checking for both the Link regex and Tooltip regex in the string for every string. Could you move this check inside the function?

Since the function automatically stops calculation if it does not find a regex match,
```js
while ((match = regex.exec(part))) { ...
```
why not just let it do both the checks inside the `injectComponent` function?

Reply:
Hi @shaavan, I'm assuming you're referring to why I moved this logic to a new function called `injectComponent`?
There is a lot of specifics involved but the general explanation is that I did it this way because the alternative is something along the lines of this:
```
function t() {
  const result = []
  
  while ((match = linkRegex.exec(part))) {
    // do link related stuff, add to result while preserving original structure of the translation
  })

  while ((match = tooltipRegex.exec(part))) {
    // do tooltip related stuff, add to result while preserving original structure of the translation
  })
  
  // trickery to get all of it into a single array, which React knows how to render.

  return result
}
```

In this approach a lot of code will have to be repeated inside the while loop. I'm assuming we'll add more types besides the `Tooltip` and `Link` down the road, so instead of having to define a while loop for every new type, I chose to create a helper function (`injectComponent`) which should make it easier to add new ones.

I'll admit this is not the cleanest code but I plan on improving this a bit more later on as the requirements become clearer.

Reply:
Thanks for the explanation, @afterburn.

I was thinking that calling the `injectComponent` function might become repetitive as the number of `Component.Type` increases. But then we can move repetitive calling under a simple _for loop_ to save us some steps.